src/compiler/checker.ts

```diff
@@ -765,7 +765,22 @@ namespace ts {
             }
         }
 
-        function getJsxNamespace(): __String {
+        function getJsxNamespace(location: Node | undefined): __String {
+            if (location) {
+                const file = getSourceFileOfNode(location);
+                if (file) {
+                    if (file.localJsxNamespace) {
+                        return file.localJsxNamespace;
+                    }
+                    const jsxPragma = file.pragmas.get("jsx");

```

Comment:
why is it `jsx` and not `jsxNamespace` like the compiler option?

Reply:
Babel supports `jsx` and I figured we'd want to support the same thing - we can alias `jsxNamespace`, if you'd like?